operator/es_client.go

```diff
@@ -365,8 +365,15 @@ func (c *ESClient) waitForEmptyEsNode(ctx context.Context, pod *v1.Pod) error {
 				default:
 					// Process response as normal if context is not done.
 					var shards []ESShard
-					err := json.Unmarshal(r.Body(), &shards)
+					body := r.Body()
+					if c.logger().Logger.Level >= log.DebugLevel {
```

Comment:
Why is this if condition needed? I think it can be left out and the log will only show up in debug mode anyway.

Reply:
Not happy about the code but some optimization. We just tried to address the string conversion as it will happen even if the debug log is not printing.  - https://github.com/zalando-incubator/es-operator/pull/422#discussion_r1634639064

Reply:
I would not do such optimizations tbh. This is no where close to a hot path :)

Reply:
Don't mind much. Can remove it.

Reply:
Done.